mmv1/products/networksecurity/SecurityProfile.yaml

```diff
@@ -0,0 +1,176 @@
+    vars:
+      resource_name: 'my-security-profile'
+    test_env_vars:
+      org_id: :ORG_ID
+custom_code:
+  !ruby/object:Provider::Terraform::CustomCode # Calling custom operation that are different from other network security resources.
+  pre_create: templates/terraform/pre_create/network_security_security_profile_operation.go.erb
+  pre_update: templates/terraform/pre_update/network_security_security_profile_operation.go.erb
+  pre_delete: templates/terraform/pre_delete/network_security_security_profile_operation.go.erb
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'name'
+    required: true
+    immutable: true
+    url_param_only: true
+    description: |
+      The name of the security profile resource.
+  - !ruby/object:Api::Type::String
+    name: 'location'
+    default_value: 'global'
+    description: |
+      The location of the security profile.
+      The default value is `global`.
+    url_param_only: true
+  - !ruby/object:Api::Type::String
+    name: 'parent'
+    description: |
+      The name of the parent this security profile belongs to.
+      Format: organizations/{organization_id}.
+    immutable: true
+    url_param_only: true
+properties:
+  - !ruby/object:Api::Type::String
+    name: 'selfLink'
+    description: |
+      Server-defined URL of this resource.
+    output: true
+  - !ruby/object:Api::Type::Time
+    name: 'createTime'
+    description: Time the security profile was created in UTC.
+    output: true
+  - !ruby/object:Api::Type::Time
+    name: 'updateTime'
+    description: Time the security profile was updated in UTC.
+    output: true
+  - !ruby/object:Api::Type::String
+    name: 'etag'
+    output: true
+    description: |
+      This checksum is computed by the server based on the value of other fields,
+      and may be sent on update and delete requests to ensure the client has an up-to-date
+      value before proceeding.
+  - !ruby/object:Api::Type::String
+    name: 'description'
+    description: |
+      An optional description of the security profile. The Max length is 512 characters.
+  - !ruby/object:Api::Type::KeyValueLabels
+    name: 'labels'
+    description: |
+      A map of key/value label pairs to assign to the resource.
+  - !ruby/object:Api::Type::NestedObject
+    name: threatPreventionProfile
+    description: The threat prevention configuration for the security profile.
+    properties:
+      - !ruby/object:Api::Type::Array
+        name: severityOverrides
+        description: The configuration for overriding threats actions by severity match.
+        item_type: !ruby/object:Api::Type::NestedObject
+          properties:
+            - !ruby/object:Api::Type::Enum
+              name: 'action'
+              description: Threat action override.
+              required: true
+              values:
+                - :ALERT
+                - :ALLOW
+                - :DEFAULT_ACTION
+                - :DENY
+                - :THREAT_ACTION_UNSPECIFIED
```

Comment:
Generally we don't need the "unspecified" enum values as an explicit Terraform option, so they can most likely be removed here and in the other enums.

Reply:
Makes sense. Done.